d2parser/parse.go

```diff
@@ -27,24 +35,43 @@ type ParseOptions struct {
 // The map may be compiled via Compile even if there are errors to keep language tooling
 // operational. Though autoformat should not run.
 //
-// If UTF16Mode is true, positions will be recorded in UTF-16 codeunits as required by LSP
+// If UTF16Pos is true, positions will be recorded in UTF-16 codeunits as required by LSP
 // and browser clients. See
 // https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocuments
 // TODO: update godocs
-func Parse(path string, r io.RuneReader, opts *ParseOptions) (*d2ast.Map, error) {
+func Parse(path string, r io.Reader, opts *ParseOptions) (*d2ast.Map, error) {
 	if opts == nil {
 		opts = &ParseOptions{
-			UTF16: false,
+			UTF16Pos: false,
 		}
 	}
 
 	p := &parser{
-		path:   path,
-		reader: r,
+		path: path,
 
-		utf16: opts.UTF16,
-		err:   opts.ParseError,
+		utf16Pos: opts.UTF16Pos,
```

Comment:
so `opts.UTF16Pos` is never actually used then? if detection is 100%, then we can get rid of CompilerOpts.UTF16Pos and ParserOpts.UTF16Pos and detect every time.

but if it's not absolute, we should only run detection and change if UTF16Pos is passed in as false

Reply:
can you get the changelog line while you're at it

Reply:
ohh lol